go/libkb/json_test.go

```diff
@@ -144,3 +145,78 @@ func TestJsonSetAndGetNull(t *testing.T) {
 	isRet = reader.GetNullAtPath(path)
 	require.True(tc.T, isRet)
 }
+
+func TestJsonTxRollback(t *testing.T) {
```

Comment:
Did this fail with the existing system?

Reply:
yes - we started with similar test when @joshblum and I were first investigating this issue. JSONFile would retain changed data after `tx.Abort()` was called.

Reply:
As I said in chat, that doesn't really matter since logout happens after any provisioning error.